Server/src/server/dataCenter/models/card/Card.java

```diff
@@ -144,4 +144,14 @@ public void decreaseRemainingNumber() {
     public void addSpell(Spell spell) {
         spells.add(spell);
     }
+
+    public boolean checkIfSameIDs(Card card){
+        int thisCardIDLength = this.getCardId().length();
+        String thisCardID = this.getCardId().substring(0, thisCardIDLength - 2);
```

Comment:
The card Ids here are the "account_name_cardID_number"  yes?

2 is a magic number here.  Magic numbers are bad,  Either leave a comment explaining what this 2 is for (I presume its about ignoring the "_<digit>" at the end of an Id). Or define the 2 as a variable (with the name explaining what it does).    

This is also a function that could be unit tested.   If someone changes how the account card Ids work, we could be left with a hard to track bug track replace bug.  But if you add a decent unit test, if anyone changes how id's work the test will fail.

Reply:
fixed